share/dkg/structs.go

```diff
@@ -227,7 +236,66 @@ func (j *JustificationBundle) Index() Index {
 	return j.DealerIndex
 }
 
-type packet interface {
+func (j *JustificationBundle) Sig() []byte {
+	return j.Signature
+}
+
+// Packet is the interface that implements the three messages that this
+// implementation uses during the different phases. This interface allows to
+// verify a DKG packet without knowing its specific type.
+type Packet interface {
 	Hash() []byte
 	Index() Index
+	Sig() []byte
+}
+
+// VerifyPacketSignature returns an error if the packet has an invalid
+// signature. The signature is verified via the information contained in the
+// config, namely the old and new nodes public keys.
+func VerifyPacketSignature(c *Config, p Packet) error {
+	if c.SkipSignatureVerification {
```

Comment:
skipping this based on a config key seems like the wrong level of granularity. I'm worried about concurrency, issues - is it possible to see new packets that you do need to verify at the same time as you're seeing repeated packets? it seems like it. as such, it seems like the modulation should happen in the call tree to this method - the callers making the call should not make the call if they're being run in a 'duplicate' context, and should be explicitly told if they're being run in that duplicate context.